Allow `clean` option to be specified with `extend`

dhritzkiv opened the pull request:
Previously, `clean` could only be set when creating an instance of `ampersand-form-view`, not when extending the base view.

wraithgar commented:
Looks good, would be nice to have a quick test for this.

dhritzkiv commented:
@wraithgar Good call. Just added some tests. but it looks like they majorly conflict with PR #30 (which I had just noticed). Perhaps commit ddd6b03 gets skipped from getting merged into master, and I can update the tests based on #30, when that's merged?

wraithgar commented:
@dhritzkiv I merged pr #30 for you.  It doesn't look like this conflicts now? Unless I'm missing something.

dhritzkiv commented:
Updated the tests so the `clean` test when extending the constructor is defined in "./tests/callbacks" and reverted "./tests/basic.js" to the way it was.

ruiramos commented:
+1, I was just about to do this :)

wraithgar commented:
Ok this is ready for a second +1 from another @AmpersandJS/core-team and we're good to go (#34 is much easier to read the diff fwiw)

dhritzkiv commented:
Sounds good. I mucked up the merge order of #30 into my branch, which is why it likely why there were conflicts when trying to merge `extend-clean` into `master`.

Thanks.
